GVFS/GVFS.Virtualization/FileSystem/FileSystemVirtualizer.cs

```diff
@@ -202,6 +202,13 @@ protected void OnWorkingDirectoryFileOrFolderDeleteNotification(string relativeP
                         this.FileSystemCallbacks.OnFolderDeleted(relativePath);
                     }
                 }
+                else
+                {
+                    // During a git command if it deletes a folder we need to track that as a tombstone

```

Comment:
nit: I might consider re-organizing these blocks so the positive case comes 1st... but this might just fit better with how I read code - as it, it works fines.

Reply:
Yeah sometimes that causes the diff to be uglier so I will leave as is for now.